src/Tasks/FileState.cs

```diff
@@ -101,6 +101,9 @@ public FileDirInfo(string filename)
                 {
                     if (NativeMethodsShared.IsWindows)
                     {
+                        // GetFileAttributesEx() will fail with long paths if not normalized
+                        _filename = NativeMethodsShared.HasMaxPath ? _filename : FileUtilities.NormalizePath(_filename);
```

Comment:
Judging by the comment, shouldn't the path get normalized only when there are max path limits?